tests/test_jira.py

```diff
@@ -0,0 +1,422 @@
+        self.assertEqual(issues[2]['updated_on'], 1457006245)
+
+    @httpretty.activate
+    def test_fetch_from_date(self):
+        """Test whether a list of issues is returned from a given date"""
+
+        from_date = str_to_datetime('2015-01-01')
+
+        bodies_json = read_file('data/jira/jira_issues_page_2.json')
+
+        httpretty.register_uri(httpretty.GET,
+                               JIRA_SEARCH_URL,
+                               body=bodies_json, status=200)
+
+        jira = Jira(JIRA_SERVER_URL)
+
+        issues = [issue for issue in jira.fetch(from_date)]
+
+        expected_req = {
+                            'expand': ['renderedFields,transitions,operations,changelog'],
+                            'jql': [' updated > 1420070400000'],
+                            'startAt': ['0']
+                        }
+
+        self.assertEqual(len(issues), 1)
+
+        self.assertEqual(issues[0]['perceval_version'], '0.1.0')
+        self.assertEqual(issues[0]['backend_version'], '0.2.0')
+        self.assertEqual(issues[0]['origin'], 'http://example.com')
+        self.assertEqual(issues[0]['backend_name'], 'Jira')
+        self.assertEqual(issues[0]['uuid'], '2e988d555915991228d81144b018c8321d628265')
+        self.assertEqual(issues[0]['updated_on'], 1457006245)
+
+        request = httpretty.last_request()
+        self.assertEqual(request.method, 'GET')
+        self.assertRegex(request.path, '/rest/api/2/search')
+        self.assertDictEqual(request.querystring, expected_req)
+
+    @httpretty.activate
+    def test_fetch_empty(self):
+        """Test whethet it works when no issues are fetched"""
+
+        bodies_json = read_file('data/jira/jira_issues_page_empty.json')
+
+        httpretty.register_uri(httpretty.GET,
+                               JIRA_SEARCH_URL,
+                               body=bodies_json, status=200)
+
+        jira = Jira(JIRA_SERVER_URL)
+
+        issues = [issue for issue in jira.fetch()]
+
+        expected_req = {
+                            'expand': ['renderedFields,transitions,operations,changelog'],
+                            'jql': [' updated > 0'],
+                            'startAt': ['0']
+                        }
+
+        self.assertEqual(len(issues), 0)
+
+        request = httpretty.last_request()
+        self.assertEqual(request.method, 'GET')
+        self.assertRegex(request.path, '/rest/api/2/search')
+        self.assertDictEqual(request.querystring, expected_req)
+
+
+class TestJiraBAckendCache(unittest.TestCase):
+    """Jira backend tests using a cache"""
+
+    def setUp(self):
+        self.tmp_path = tempfile.mkdtemp(prefix='perceval_')
+
+    def tearDown(self):
+        shutil.rmtree(self.tmp_path)
+
+    @httpretty.activate
+    def test_fetch_from_cache(self):
+        """Test whether a list of issues is returned from cache"""
+
+        bodies_json = read_file('data/jira/jira_issues_page_2.json')
```

Comment:
Why don't you use the same data sets that you use on `test_fetch()`? It's more easier to understand what we're doing with the test and includes a full case.